### ISSUE TITLE
positive? vs >0 cops

### ISSUE THREAD

**Issue by Ana06** [bbatsov/rubocop#5564]
Add a new cop to ensure that [`positive?`](https://ruby-doc.org/core-2.5.0/Numeric.html#method-i-positive-3F) is used instead of `> 0` or the other way around.

I also opened and issue in the Style guide: https://github.com/bbatsov/ruby-style-guide/issues/701

What should be the default? I personally do not see a reason to use this `positive?` method, as `> 0` was already clear, but maybe someone come up with a reason to use it.

**Comment by mikegee** [comment]
I think we have this. https://rubocop.readthedocs.io/en/latest/cops_style/#stylenumericpredicate

But, it doesn't work somehow?

My little test file makes no offense

```ruby
if foo > 0
  :positive
end
```

even though it claims to be enabled

```
bundle exec bin/rubocop --show-cops Style/NumericPredicate
# Supports --auto-correct
Style/NumericPredicate:
  Description: Checks for the use of predicate- or comparison methods for numeric comparisons.
  StyleGuide: "#predicate-methods"
  AutoCorrect: false
  Enabled: true
  EnforcedStyle: predicate
  SupportedStyles:
  - predicate
  - comparison
  Exclude:
  - "/app/spec/**/*"
```

Weird.

**Comment by Drenmi** [comment]
What's your configured `TargetRubyVersion` @mikegee? These predicates were only introduced in Ruby 2.3.

**Comment by Drenmi** [comment]
> Add a new cop to ensure that positive? is used instead of > 0 or the other way around.

We already have this. 🙂 

> What should be the default? I personally do not see a reason to use this positive? method, as > 0 was already clear, but maybe someone come up with a reason to use it.

The default is the predicate. The benefit of that is you'll avoid silent errors when accidentally comparing to `nil`, and will instead get a runtime error. The downside is people might not immediately know whether 0 is considered positive or not.

**Comment by mikegee** [comment]
> What's your configured `TargetRubyVersion`?

I reproduce Rubocop issues in a subdirectory, and Rubocop's config has `TargetRubyVersion: 2.1`. Oops

**Comment by Ana06** [comment]
@Drenmi ok, sorry, I didn't know it 🙈 

But I think that it is not in the Ruby Style Guide, is it? 🤔
